app/serializers/workflow_serializer.rb

```diff
@@ -22,7 +22,7 @@ class WorkflowSerializer
   preload :subject_sets, :attached_images
 
   def version
-    "#{@model.current_version_number}.#{content_version}"
+    "#{@model.major_version}.#{content_version}"
```

Comment:
Was the `current_version_number` the source of the errors as it relied on the paper tail array index version (offset by 1)?

Should the code in `CacheModelVersion` and `ModelVersion` etc be removed (another PR)?

Reply:
Yes, that code is slated for removal already in a different branch. This just brings forward this line of diff since we've now found it actually fixes an issue.